editor/src/clj/editor/code/view.clj

```diff
@@ -3544,11 +3494,15 @@
 
 (defn- make-view! [graph parent resource-node opts]
   (let [{:keys [^Tab tab app-view grammar open-resource-fn project]} opts
+        basis (g/now)
+        resource-node-type (g/node-type* basis resource-node)
+        editable (and (g/has-property? resource-node-type :modified-lines)
+                      (g/connected? basis resource-node :save-data project :save-data))
```

Comment:
* The `ResourceNode` must have a `modified-lines` property to be editable using the `CodeEditorView`. This was always the case, even before this PR.
* Any `ResourceNode` that represents an editable project file will be connected to the `save-data` input, but nodes coming from library dependencies or builtins will not.